common/components/controllers/CreateEventProjectController.jsx

```diff
@@ -0,0 +1,207 @@
+    this.state = {
+      eventId: eventId,
+      projectId: projectId,
+      waitForLoad: !!projectId,
+      startStep: 0,
+      steps: [
+        {
+          header: "Let's submit your project to the hackathon",
+          subHeader: "",
+          onSubmit: this.onSubmit,
+          onSubmitSuccess: this.onNextPageSuccess,
+          formComponent: CreateEventProjectSelect,
+        },
+        {
+          header: "Let others know what your hackathon project is about",
+          subHeader:
+            "Share your internal resources to help volunteers understand your goals and processes. " +
+            "Any changes below will not impact the details of your long-term project profile.",
+          onSubmit: this.onSubmit,
+          onSubmitSuccess: this.onNextPageSuccess,
+          formComponent: CreateEventProjectScope,
+        },
+        {
+          header: "What resources would you like to share?",
+          subHeader:
+            "Share your internal resources to help volunteers understand the goals and processes. " +
+            "Any changes below will not impact the details of your long-term project profile.",
+          onSubmit: this.onSubmit,
+          onSubmitSuccess: this.onNextPageSuccess,
+          formComponent: EventProjectResourcesForm,
+        },
+        {
+          header: "What type of volunteers does your hackathon team need?",
+          subHeader:
+            "You can always change the type of help your team needs later. " +
+            "Any changes below will not impact the details of your long-term project profile.",
+          submitButtonText: "Submit",
+          onSubmit: this.onSubmit,
+          onSubmitSuccess: this.onFinalSubmitSuccess,
+          formComponent: EventProjectPositionsForm,
+        },
+      ],
+    };
+  }
+
+  static getStores(): $ReadOnlyArray<FluxReduceStore> {
+    return [FormFieldsStore];
+  }
+
+  static calculateState(prevState: State, props: Props): State {
+    let state: State = _.clone(prevState) || {};
+    const project: MyProjectData = FormFieldsStore.getFormFieldValue("project");
+    if (project) {
+      state.projectId = project?.project_id;
+    }
+    return state;
+  }
+
+  componentDidMount(): void {
+    if (this.state.projectId) {
+      this.setState(
+        { isLoading: true },
+        EventProjectAPIUtils.fetchEventProjectDetails(
+          this.state.eventId,
+          this.state.projectId,
+          this.loadEventProjectDetails.bind(this),
+          this.handleLoadProjectError.bind(this)
+        )
+      );
+    }
+  }
+  updatePageUrl() {
+    if (this.state.projectId && !url.argument("project_id")) {
+      url.updateArgs({ project_id: this.state.projectId });
+    }
+    utils.navigateToTopOfPage();
+  }
+
+  loadEventProjectDetails(eventProject: EventProjectAPIDetails): void {
+    if (!CurrentUser.isCoOwnerOrOwner(eventProject) && !CurrentUser.isStaff()) {
+      // TODO: Handle someone other than owner
```

Comment:
todo on handling loads for non-owners?

Reply:
Recorded as "Show error for people who are not the project owner or admin"